lib/netext/httpext/request.go

```diff
@@ -162,9 +162,90 @@ func compressBody(algos []CompressionType, body io.ReadCloser) (io.Reader, int64
 	return buf, int64(buf.Len()), contentEncoding, body.Close()
 }
 
+func readResponseBody(
+	state *lib.State, respType ResponseType, resp *http.Response, respErr error,
+) (interface{}, error) {
+
+	if resp == nil || respErr != nil {
+		return nil, respErr
+	}
+
+	// Transperently decompress the body if it's has a content-encoding we
+	// support. If not, simply return it as it is.
+	contentEncoding := strings.TrimSpace(resp.Header.Get("Content-Encoding"))
+	if compression, err := CompressionTypeString(contentEncoding); err == nil {
+		switch compression {
+		case CompressionTypeDeflate:
+			resp.Body, respErr = zlib.NewReader(resp.Body)
+		case CompressionTypeGzip:
+			resp.Body, respErr = gzip.NewReader(resp.Body)
+		default:
+			// We have not implemented a compression ... :(
+			respErr = fmt.Errorf(
+				"unsupported compression type %s for decompression - this is a bug in k6, please report it",
+				compression,
+			)
+		}
+	}
+
+	if respType == ResponseTypeNone {
+		_, err := io.Copy(ioutil.Discard, resp.Body)
+		_ = resp.Body.Close()
+		if err != nil && err != io.EOF {
+			respErr = err
+		}
+		return nil, respErr
+	}
+
+	buf := state.BPool.Get()
+	buf.Reset()
+	_, err := io.Copy(buf, resp.Body)
+	_ = resp.Body.Close()
+	if err != nil && err != io.EOF {
+		respErr = err
+	}
+
+	var result interface{}
+	// Binary or string
+	switch respType {
+	case ResponseTypeText:
+		result = buf.String()
+		state.BPool.Put(buf)
+	case ResponseTypeBinary:
+		result = buf.Bytes()
+		// We don't return the buffer to the pool, because the byte
+		// slice could still point to its underlying slice.
+	default:
+		respErr = fmt.Errorf("unknown responseType %s", respType)
+		state.BPool.Put(buf)
+	}
+
+	return result, respErr
+}
+
+func updateK6Response(k6Response *Response, finishedReq *finishedRequest) {
+	k6Response.ErrorCode = int(finishedReq.errorCode)
+	k6Response.Error = finishedReq.errorMsg
+	trail := finishedReq.trcerTrail
+
+	if trail.ConnRemoteAddr != nil {
+		remoteHost, remotePortStr, _ := net.SplitHostPort(trail.ConnRemoteAddr.String())
```

Comment:
I am not certain but ... maybe cast ConnRemoteAddr to [net.TCPAddr](https://golang.org/pkg/net/#TCPAddr) and just get the values from inside ? maybe just change ConnRemoteAddr to be `net.TCPAddr`? 

I know you are just copying code but maybe this is worth a try ?

Reply:
Decided against this, since it's not worth the risk for type assert stuff here - it will probably break when http3 gets added, since it's going to use UDP anyway :smile:

Reply:
Well... I would say that the type assert  should be earlier when we put it into the `ConnRemoteAddr` and that we should probably just do all the parsing there and have the `ConnRemoteAddr` be some structure that we has host and port as fields. 
And we can arguably type switch on it and if it is TCPAddr or UDPAddr handle it faster, but I agree that it might not be worth the effort for some net.SplitHostPort and strconv calls